### PR TITLE
Many small updates.

### PR DESCRIPTION
**Pull request by vext01**
See commit messages.

This is a draft, because we need bencher13 back before we merge, or the new docs won't deploy.

### CODE REVIEW

File: src/misc/ci-cycles.md [hunk 1]
```diff
@@ -41,29 +39,26 @@ Next the reviewer switches to the `ykrustc` pull request and conducts a review
...
-Next the reviewer should ask the PR author to push a "cycle breaker" commit to
-`ykrustc`. This commit overrides the compiler's dependency on `ykpack`, making
-it instead use the commit hash for the head of the branch we reviewed in stage
-2.
+4. **Prime the CI**
 
-[Here's an example cycle breaker commit](https://github.com/softdevteam/ykrustc/commit/abd1c2b7669c4ab6be8f9a9e6c1704a7e70c2088)
+Next the reviewer should ask the PR author to "prime the CI". A commit should
+be pushed which overrides the yk dependency to the branch reviewed in stage 2
+of this document. This commit should override the compiler's dependences on
```
Comment (ptersilie) [inline src/misc/ci-cycles.md:46]: s/dependences/dependencies/

File: src/misc/ci-cycles.md [hunk 2]
```diff
@@ -41,29 +39,26 @@ Next the reviewer switches to the `ykrustc` pull request and conducts a review
...
-the branch we reviewed in step 2 and CI will use the newly cached compiler to
-build it, before merging the branch.
+`ykrustc` which uses the new `ykpack`. Now the reviewer can continue to invoke
+bors on the branch we reviewed in step 2 and CI will use the newly cached
+compiler to build it, before merging the branch.
+
+6. **Merge ykrustc**
 
-7. **Undo the cycle breaker commit**
+The reviewer can now ask the PR author remove the priming commit introduced
```
Comment (ptersilie) [inline src/misc/ci-cycles.md:61]: > the PR **to** author

File: src/misc/ci-cycles.md [hunk 3]
```diff
@@ -41,29 +39,26 @@ Next the reviewer switches to the `ykrustc` pull request and conducts a review
...
+at stage 4 (using a force push) and ask the reviewer to run bors.
```
Comment (ptersilie) [inline src/misc/ci-cycles.md:62]: and once this is done, the PR can be merged with bors.

File: src/tech/irs.md [hunk 4]
```diff
@@ -32,3 +34,16 @@ TIR is basically SIR with guards instead of branches. TIR is the basis for a
...
+## Local Variables
+
+MIR, SIR and TIR all have the notion of local variables. Each local variable is
+identified by an integer (its *index*). These indices are unique only within
+the scope of a function.
+
+At the time of writing, there is a one-to-one relationship between the variable
+indices in MIR and SIR, however this may change in the future.
+
+At the time a TIR trace is compiled, variables are renamed so that (e.g.)
```
Comment (ptersilie) [inline src/tech/irs.md:47]: > At the time a TIR trace is ~~compiled~~ generated

Comment (ptersilie) [reply src/tech/irs.md:47]: Or better:
> Before a TIR trace is compiled, variables are...

File: src/tech/pipeline.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+This means that (due to ASLR) these addresses are unique to a particular run of
+the program.
+
+## SIR Traces
+
+Given a PT trace, Yorick then constructs a SIR trace. A SIR trace is a list of
+<symbol-name, block-index> pairs. Each such pair identifies a SIR block that
+execution passed through during tracing.
+
+To make a SIR trace from a PT trace, Yorick uses the so-called "mapper". When
```
Comment (ptersilie) [inline src/tech/pipeline.md:24]: Maybe:

...Yorick uses the [HWTMapper](https://github.com/softdevteam/yk/blob/master/yktrace/src/hwt/mapper.rs)

**Comment by vext01** [comment]
Comments addressed. OK to squash?

**Comment by ptersilie** [comment]
Looks good. Please squash.

**Comment by vext01** [comment]
splat

**Comment by vext01** [comment]
Fixed conflicts. Not sure what happened there.

Ready for bors, I believe.

**Pull request merged**